software/apache/README.md

```diff
@@ -25,171 +21,144 @@ This guide walks you through setting up Apache natively on your MacOS with the a
-    #LoadModule php7_module /usr/local/opt/php@7.2/lib/httpd/modules/libphp7.so
+## Installing PHP
 
-**Edit the following lines:**
+The following script will install each version of PHP, with FPM, configure
+their listening ports for 90## (9071, for example), and configure
+FPM to start using brew services.
 
-    <IfModule dir_module>
-      DirectoryIndex index.html
-    </IfModule>
+    php_versions=("5.6" "7.0" "7.1" "7.2")
 
-**To the following:**
+    for version in ${php_versions[*]}; do
+      formula="php@$version"
 
-    <IfModule dir_module>
-      DirectoryIndex index.php index.html
-    </IfModule>
+      # Install php version
+      brew install "$formula"
 
-    <FilesMatch \.php$>
-      SetHandler application/x-httpd-php
-    </FilesMatch>
+      # Configure each PHP-FPM (FastCGI Process Manager) listening port
+      www_conf_path="/usr/local/etc/php/$version/php-fpm.d/www.conf"
+      fpm_conf_path="/usr/local/etc/php/$version/php-fpm.conf"
+      conf_path="unknown file"
+      port_suffix=${version/\./}
 
-### PHP Switcher Script
+      if grep -q "listen = 127.0.0.1:90\d\d" "$www_conf_path" 2>/dev/null; then
+        conf_path="$www_conf_path"
+      elif grep -q "listen = 127.0.0.1:90\d\d" "$fpm_conf_path"; then
+        conf_path="$fpm_conf_path"
+      fi
 
-    $ curl -L https://gist.githubusercontent.com/rhukster/f4c04f1bf59e0b74e335ee5d186a98e2/raw > /usr/local/bin/sphp
-    $ chmod +x /usr/local/bin/sphp
+      if [ -f "$conf_path" ]; then
+        echo "Updating listen port to $port_suffix in $conf_path"
+        sed -i.bak "s/listen = 127\.0\.0\.1\:90[[:digit:]][[:digit:]]/listen = 127.0.0.1:90$port_suffix/g" "$conf_path"
+      else
+        echo "Unable to locate fpm conf file for $version $conf_path"
+      fi
 
-Test PHP switcher by running
+      # Configure PHP-FPM to auto start using Homebrew services
+      brew services start "$formula"
+    done
 
-    $ sphp 7.0
+Important file locations:
 
-This should uncomment out the php 7.0 `LoadModule` line in `/usr/local/etc/httpd/httpd.conf` and make your shell's `php -v` report 7.0.*
-### Vhosts and the Hosts File
+    /usr/local/etc/php/5.6/php.ini
+    /usr/local/etc/php/7.0/php.ini
 
-Using Apache 2.4  you should be able to add vhosts using the `vhost` directory.
+## Setting up a new PHP site in Apache
 
-In your local apache `/usr/local/etc/httpd/` directory run the following commands.
+### Create the vHost file
+Each site will have it's own _vhost_ file in `/usr/local/etc/httpd/vhosts`.
 
-    $ mkdir vhosts
-    $ cd vhosts
-    $ touch <your_project_name>.conf
+:warning: You’ll need to swap out the following placeholders with your own settings:
 
-Open the file `<your_project_name>.conf`
+  * `<project_name>`: Domain safe name such as `seesparkbox`
+  * `<project_root>`: Web root such as `/projects/seesparkbox/static/dist`
+  * `<php_port>`: Port to the PHP-FPM instance, such as `9071` for 7.1
```

Comment:
This comment will make more sense if the PHP install script above explains the port setup more clearly. See my comment in that section.